### PR TITLE
Implement component type

### PR DESCRIPTION
**Pull request by KoreanThinker**
I simply defined the type.

```ts
export interface ReadMoreProps {
    style?: StyleProp<TextStyle>;
    seeMoreStyle?: StyleProp<TextStyle>;
    seeLessStyle?: StyleProp<TextStyle>;
    wrapperStyle?: StyleProp<ViewStyle>;
    children?: React.ReactNode;
    numberOfLines?: number;
    seeMoreText?: string;
    seeLessText?: string;
    animate?: boolean;
    customTextComponent?: React.ReactNode;
    ellipsis?: string;
    allowFontScaling?: boolean;
    onExpand?: () => void;
    onCollapse?: () => void;
    expandOnly?: boolean;
    seeMoreOverlapCount?: number;
    debounceSeeMoreCalc?: number;
    onLayout?: (event: LayoutChangeEvent) => void;
    onTextLayout?: (event: NativeSyntheticEvent<TextLayoutEventData>) => void;
    onReady?: () => void;
}
```

### CODE REVIEW

**Comment by KoreanThinker** [comment]
@fawaz-ahmed Please review it.

**Comment by fawaz-ahmed** [comment]
@KoreanThinker this is a great addition. Thanks for supporting.
Can you also change version in package.json to 2.3.0 ?

File: package.json [hunk 1]
```diff
@@ -37,6 +38,10 @@
   "dependencies": {
     "prop-types": "^15.7.2"
   },
+  "devDependencies": {
+    "@types/react-native": "^0.66.0",
+    "react": "^16.9.0"
```
Comment (fawaz-ahmed) [inline package.json]: Is this package react needed ?

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "2.2.2",
```
Comment (fawaz-ahmed) [inline package.json]: Can you also bump version to 2.3.0 ?

File: package.json (hunk 1 above)
Comment (fawaz-ahmed) [reply package.json]: Since its already installed with every react native app

File: package.json [hunk 3]
```diff
@@ -37,6 +38,10 @@
   "dependencies": {
     "prop-types": "^15.7.2"
   },
+  "devDependencies": {
+    "@types/react-native": "^0.66.0",
```
Comment (fawaz-ahmed) [inline package.json]: This package is not being used anywhere or am I mistaken. Please check if not then let's remove it

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { StyleProp, TextStyle, ViewStyle, LayoutChangeEvent, NativeSyntheticEvent, TextLayoutEventData } from 'react-native';
+
+export interface ReadMoreProps {
+    style?: StyleProp<TextStyle>;
+    seeMoreStyle?: StyleProp<TextStyle>;
+    seeLessStyle?: StyleProp<TextStyle>;
+    wrapperStyle?: StyleProp<ViewStyle>;
+    children?: React.ReactNode;
```
Comment (fawaz-ahmed) [inline index.d.ts:9]: Children can be a node or multiple nodes.

**Review by fawaz-ahmed — CHANGES_REQUESTED** [review]
PR looks good and this will help alot moving towards ts. @KoreanThinker just think that the devDependencies are not needed.
Please see comments.

File: index.d.ts (hunk 4 above)
Comment (KoreanThinker) [reply index.d.ts:9]: @fawaz-ahmed React.ReactNode can be contain multiple nodes
This code working well.
```tsx
<ReadMore>
   <Text>hi</Text>
   <Icon />
</ReadMore>
```

**Review by fawaz-ahmed — APPROVED** [review]

**Pull request merged**